docker/utils/utils.py

```diff
@@ -615,6 +615,16 @@ def create_container_config(
         if volumes_from is not None:
             raise errors.InvalidVersion(message.format('volumes_from'))
 
+    # NetworkMode must be present and valid in host config from 1.20 onwards
```

Comment:
Well, the user can decide to provide their own dictionary for `host_config` instead of using `create_host_config`, but at that point they are making the conscious choice of not perusing the library's validation and transformation features the library provides, which I believe docker-py should let them do without tampering. AFAIK the tests always use `create_host_config`.

The only drawback is that `create_host_config` doesn't have access to the client's API version. Not sure we have a way to work around that.